### PR TITLE
Mention Groovy 3 in release notes

### PR DESCRIPTION
**Pull request by bmuskalla**

### CODE REVIEW

File: subprojects/docs/src/docs/release/notes.md [hunk 1]
```diff
@@ -62,6 +62,11 @@ File system watching and configuration caching is enabled for the comparison.
 
 You can find the performance test project [here](https://github.com/gradle/santa-tracker-performance).
 
+## Groovy 3 upgrade
+
+Gradle has been upgraded to use Groovy 3. Groovy 3 comes with a brand new parser (code-named Parrot) and host of other new features and capabilities. In order to learn more about the improvements and new features in Groovy, please refer to [their release notes](https://groovy-lang.org/releasenotes/groovy-3.0.html).
```
Comment (pioterj) [inline subprojects/docs/src/docs/release/notes.md:67]: @bmuskalla Why should our users care about the new Groovy parser and its code name?

Comment (bmuskalla) [reply subprojects/docs/src/docs/release/notes.md:67]: The new Groovy parser will be the source of almost all issues we expect as it has slightly different behavior and is used to parse all build scripts. So it's important for people to understand that there is a significant change. If you feel strong about removing the codename, that's fine by me. Can do that

Comment (pioterj) [reply subprojects/docs/src/docs/release/notes.md:67]: What is explain makes sense but does not come across in the release notes though that mention the new parser without additional context and next to new features and capabilities without explaining how they are relevant to Gradle users.

Comment (bmuskalla) [reply subprojects/docs/src/docs/release/notes.md:67]: In theory, it should just work. And maybe if this is not clear: This is not our final release notes section on Groovy3. We wanted something simple for the milestone and we'll expand this with proper hints for common problems we see in the wild (or make them part of the upgrade guide)

**Pull request merged**